delocate/tools.py

```diff
@@ -163,9 +176,197 @@ def parse_install_name(line):
         compatibility version
     current_version : str
         current version
-    """
+
+    Examples
+    --------
+    >>> parse_install_name("/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 905.6.0)")
+    ('/usr/lib/libc++.1.dylib', '1.0.0', '905.6.0')
+    """  # noqa: E501
     line = line.strip()
-    return IN_RE.match(line).groups()
+    match = IN_RE.match(line)
+    if not match:
+        raise ValueError(f"Could not parse {line!r}")
+    libname, compat_version, current_version = match.groups()
+    return libname, compat_version, current_version
```

Comment:
Doesn't the `match.groups()` already assert there are three entries?

```suggestion
    return match.groups()
```

Reply:
The first assert makes sure that `match` isn't `None` which casts it to a non-Optional type.  The unpacking acts as a second cast that verifies the length of the tuple.  Your suggestion will return a literal `tuple[str, ...]` type where an `tuple[str, str, str]` is expected and this will be flagged by Mypy.

I imagine that one day the static typing will become complex enough to derive the return type from the regex string, but for now I do this.